### PR TITLE
8.0 distribution list imp

### PR DESCRIPTION
**Pull request by JonathanNEMRY**
Improve feature 'more filter' by adding this option into the 'alternative_object' mode
Fix a bug related to this alternative_object
Improve test to cover those changes

### CODE REVIEW

**Comment by JonathanNEMRY** [comment]
Any news about this PR?

File: distribution_list/distribution_list.py [hunk 1]
```diff
@@ -292,8 +293,12 @@ def get_complex_distribution_list_ids(self, cr, uid, ids, context=None):
                         else:
                             result_ids .append(val[main_object])
                 if alternative_object:
-                    domain_alternative_objects = domains + \
-                        [(main_object, '!=', False)]
+                    domain_alternative_objects = \
+                        [('id', 'in', res_ids),
+                         (alternative_object, '!=', False)]
+                    if context.get('alternative_more_filter', False):
```
Comment (lmignon) [inline distribution_list/distribution_list.py]:
``` python
if context.get('alternative_more_filter')
```

File: distribution_list/distribution_list.py [hunk 2]
```diff
@@ -272,11 +274,10 @@ def get_complex_distribution_list_ids(self, cr, uid, ids, context=None):
             if dls:
                 dls_target_model = dls[0].dst_model_id.model
             if dls_target_model and res_ids:
-                domains = [('id', 'in', res_ids)]
+                domain_main_objects = [('id', 'in', res_ids),
+                                       (main_object, '!=', False)]
                 if context.get('more_filter', False):
```
Comment (lmignon) [inline distribution_list/distribution_list.py]:
``` python
if context.get('more_filter')
```

File: distribution_list/distribution_list.py [hunk 3]
```diff
@@ -292,8 +296,12 @@ def get_complex_distribution_list_ids(self, cr, uid, ids, context=None):
                         else:
                             result_ids .append(val[main_object])
                 if alternative_object:
```
Comment (lmignon) [inline distribution_list/distribution_list.py]: Is-it possible to split the computation of main_object and alternative_object into 2 sub methods to improve the readability and the modularity?

**Comment by JonathanNEMRY** [comment]
:+1:

File: distribution_list/distribution_list.py [hunk 4]
```diff
@@ -235,86 +238,69 @@ def get_ids_from_distribution_list(self, cr, uid, ids, safe_mode=True,
 
         return list(set_included_ids)
 
+    def _get_ids(self, cr, uid, ids, model, fld, dom, sort, context=None):
+        res_ids = []
```
Comment (lmignon) [inline distribution_list/distribution_list.py:249]: fld means field_name?, dom domain?

**Comment by Cedric-Pigeon** [comment]
:+1:

**Comment by lmignon** [comment]
Nice refactoring!  A description would be nice for the _get_ids method. :+1:

**Pull request merged**